### PR TITLE
Implementation of a new function called spike contrast. 

### PR DESCRIPTION
**Pull request by Broxy7**
We implemented a new function called spike_contrast which is capable of calculating the synchrony of spike trains.
I am fairly new to software testing and would appreciate any kind of feedback.
Furthermore this is my first pull request so I also would appreciate any kind of feedback.

### CODE REVIEW

**Comment by dizcza** [comment]
Hello Philipp and Manual,
very interesting, thank you.

We're currently in the process of integrating another (related?) pull request #322, where we define such thing as `complexity` class. Could you share the paper that you mentioned in the references section with us in order to verify a potential overlap of methods being implemented in parallel with #322? 

As for the code, it looks great to be the first contribution. I see that your code already have the documentation, comments, and tests. The tests are failing and pep8 is complaining, but we'll come to this later on. I'd like to understand your method first.

Danylo.

**Comment by dizcza** [comment]
Thank you for the paper. We're currently reading it and write the feedback here.

**Comment by dizcza** [comment]
After a quick look, the outlined difference with #322 is:
* your method, spike contrast, finds the bin size that maximizes the synchrony characteristic of a network (spike trains) while
* the method in #322 requires a fixed user-defined bin size, and it allows estimating the complexity (synchrony) of spike trains and removing the artifacts (identical spike times of different neurons, which is not biologically plausible).

We have meetings each Mon where we discuss the pull request, so expect more updates by that time.

**Comment by Broxy7** [comment]
Many thanks for the feedback! 
We are looking forward to hear from you next Mon.

**Comment by dizcza** [comment]
Instead of describing what is needed to be fixed, I created a [branch](https://github.com/INM-6/elephant/blob/pr/spike_contrast/elephant/spike_contrast.py) with all necessary updates. Once we deal with the todo list, I'll open a pull request from this branch to your fork for Elephant. Then you can either accept it or copy-paste the changes.

### Todo list
1. [x] Refactor the code.
2. [ ] Fix python2.7 and python3.5 test issues.
3. [ ] Move spike\_contrast to an existing module (probably, spike-train-correlation).
4. [ ] Write a test that reproduces at least one figure from the paper.

I'll do 1-3.

@Broxy7 can you write a test that reproduces the results for the Izhikevich network plot (Test data 3) in python? I noticed the [original github repository](https://github.com/biomemsLAB/Spike-Contrast), which has zipped test data. If I start doing it by myself, I'll have a lot of questions. For example, (1) does each folder `Sim-xx` in `Testdata/3-izhikevich_network` represent a trial? (2) What does a file in each of these folders represent then?

In short, it'd nice if you
1. Pack the data of Izhikevich network experiment either in .npy (platform-dependent, if you allow pickling) or .csv format (cross-platform; suited if the data size is <10 Mb) and share it with me (you can drop it here or send to contact@python-elephant.org).
2. Write a corresponding unit test for this data set. You can pick either my branch or continue on this original branch from which you started the pull request - the logic is not changed, except for I don't transpose the array.

Will you do it?

**Comment by Broxy7** [comment]
Yes I will do point 7 on the todo list.
I will get in touch with Manuel and we will reach out for you.

**Comment by dizcza** [comment]
@Broxy7 please let us know once you think the test and test data is ready for merging.

**Comment by Broxy7** [comment]
I uploaded the JSON File and the new Software test. The test ran without any errors. 
It should be ready for merging.

**Comment by dizcza** [comment]
@Broxy7 
We published the test data under CC-BY 4.0 license here: https://gin.g-node.org/INM-6/elephant-data/src/spike-contrast/dataset-3
Do you agree with the license type and the readme description? The zipped data will be downloaded each time the test runs.

**Comment by Broxy7** [comment]
@dizcza 
We accept the README description and the CC-BY 4.0 license.

Cheers
Philipp

**Comment by Broxy7** [comment]
@dizcza 
I accepted your pull-request and the final version should be ready for merging. 
Cheers,  
Philipp

**Comment by dizcza** [comment]
@Broxy7 
For some reason, GitHub could not resolve the changes you made with the master branch: it shows that 80+ files have been changed, which is not true, of course. It also shows git conflicts although there should not be any.

I understand that this problem could be hard for you to resolve manually, and, unfortunately, I cannot push into your own repository. I'll describe the solution to fix GitHub problems on Monday, Oct 5.

**Comment by dizcza** [comment]
@Broxy7 
Please do the following:
* Go to the [pull requests](https://github.com/NeuralEnsemble/elephant/pulls)
* Click "New pull request"
* Click "compare across forks"
* Choose "INM-6" branch "pr/spike_contrast" as the head and "NeuralEnsemble" branch "master" as the base, as in the screenshot below

I want you to be the initiator of the pull request. Once you open a new pull request, I'll merge it - no further changes in the code are needed, it's perfect.

After your new pull request is merged, I'll close _this_ pull request.

**Comment by Broxy7** [comment]
@dizcza 
done :)

**Comment by dizcza** [comment]
Reopened in #354
